apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/grpc/GrpcMetadataResolver.java

```diff
@@ -19,15 +19,35 @@ public class GrpcMetadataResolver {
   /** Metadata key for passing the cassandra token to the Bridge. */
   private final Metadata.Key<String> cassandraTokenKey;
 
+  /** Default metadata. */
+  private final Metadata defaultMetadata;
+
+  // TODO add validation @Pattern(regexp = "rest|graphql")  after
+  //  https://github.com/quarkusio/quarkus/issues/28783
   @Inject
-  public GrpcMetadataResolver(GrpcMetadataConfig config) {
+  public GrpcMetadataResolver(GrpcMetadataConfig config, @SourceApi String sourceApi) {
     this.tenantIdKey = Metadata.Key.of(config.tenantIdKey(), Metadata.ASCII_STRING_MARSHALLER);
     this.cassandraTokenKey =
         Metadata.Key.of(config.cassandraTokenKey(), Metadata.ASCII_STRING_MARSHALLER);
+
+    // default metadata includes source api
+    Metadata.Key<String> sourceApiKey =
+        Metadata.Key.of(config.sourceApiKey(), Metadata.ASCII_STRING_MARSHALLER);
```

Comment:
I know short-living garbage is not a big problem in general but couldn't we just create this key once (add a convenience method in `GrpcMetadataConfig` to get pre-created `Metadata.Key` instead of `String` for creating one) and reuse it? `Metadata.Key` is annotated as Immutable and hence thread-safe.

Reply:
Don't get this, this is a constructor, called only one time during init? Does exactly that by creating a single instance of the default metadata..

Reply:
@ivansenic Ok sorry. Missed that context. N/m.